### ISSUE TITLE
Exceptions when handling exceptions

### ISSUE THREAD

**Issue by shevek** [raelgc/scudcloud#544]
## ScudCloud Version

Paste the output for `scudcloud --version` below:
`ScudCloud 1.50
Python 3.5.2
Qt 5.5.1
PyQt 5.5.1
SIP 4.17
`

## Distro and Desktop info
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.2 LTS
Release:        16.04
Codename:       xenial

## Expected behavior

Does not crash

## Actual behavior

ScudCloud using 50% CPU on a Broadwell system

```
SpellCheck: Language = "en_US"
Traceback (most recent call last):
  File "/usr/lib/python3.5/urllib/request.py", line 1254, in do_open
    h.request(req.get_method(), req.selector, req.data, headers)
  File "/usr/lib/python3.5/http/client.py", line 1106, in request
    self._send_request(method, url, body, headers)
  File "/usr/lib/python3.5/http/client.py", line 1151, in _send_request
    self.endheaders(body)
  File "/usr/lib/python3.5/http/client.py", line 1102, in endheaders
    self._send_output(message_body)
  File "/usr/lib/python3.5/http/client.py", line 934, in _send_output
    self.send(msg)
  File "/usr/lib/python3.5/http/client.py", line 877, in send
    self.connect()
  File "/usr/lib/python3.5/http/client.py", line 1252, in connect
    super().connect()
  File "/usr/lib/python3.5/http/client.py", line 849, in connect
    (self.host,self.port), self.timeout, self.source_address)
  File "/usr/lib/python3.5/socket.py", line 711, in create_connection
    raise err
  File "/usr/lib/python3.5/socket.py", line 702, in create_connection
    sock.connect(sa)
TimeoutError: [Errno 110] Connection timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/scudcloud/wrapper.py", line 234, in populate
    file_name, headers = request.urlretrieve(data['icon'], icon_path)
  File "/usr/lib/python3.5/urllib/request.py", line 188, in urlretrieve
    with contextlib.closing(urlopen(url, data)) as fp:
  File "/usr/lib/python3.5/urllib/request.py", line 163, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.5/urllib/request.py", line 466, in open
    response = self._open(req, data)
  File "/usr/lib/python3.5/urllib/request.py", line 484, in _open
    '_open', req)
  File "/usr/lib/python3.5/urllib/request.py", line 444, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.5/urllib/request.py", line 1297, in https_open
    context=self._context, check_hostname=self._check_hostname)
  File "/usr/lib/python3.5/urllib/request.py", line 1256, in do_open
    raise URLError(err)


urllib.error.URLError: <urlopen error [Errno 110] Connection timed out>
"No such interface 'org.freedesktop.DBus.Properties' on object at path /"
Traceback (most recent call last):
  File "/usr/lib/python3.5/urllib/request.py", line 1254, in do_open
    h.request(req.get_method(), req.selector, req.data, headers)
  File "/usr/lib/python3.5/http/client.py", line 1106, in request
    self._send_request(method, url, body, headers)
  File "/usr/lib/python3.5/http/client.py", line 1151, in _send_request
    self.endheaders(body)
  File "/usr/lib/python3.5/http/client.py", line 1102, in endheaders
    self._send_output(message_body)
  File "/usr/lib/python3.5/http/client.py", line 934, in _send_output
    self.send(msg)
  File "/usr/lib/python3.5/http/client.py", line 877, in send
    self.connect()
  File "/usr/lib/python3.5/http/client.py", line 1252, in connect
    super().connect()
  File "/usr/lib/python3.5/http/client.py", line 849, in connect
    (self.host,self.port), self.timeout, self.source_address)
  File "/usr/lib/python3.5/socket.py", line 711, in create_connection
    raise err
  File "/usr/lib/python3.5/socket.py", line 702, in create_connection
    sock.connect(sa)
TimeoutError: [Errno 110] Connection timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/scudcloud/wrapper.py", line 234, in populate
    file_name, headers = request.urlretrieve(data['icon'], icon_path)
  File "/usr/lib/python3.5/urllib/request.py", line 188, in urlretrieve
    with contextlib.closing(urlopen(url, data)) as fp:
  File "/usr/lib/python3.5/urllib/request.py", line 163, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.5/urllib/request.py", line 466, in open
    response = self._open(req, data)
  File "/usr/lib/python3.5/urllib/request.py", line 484, in _open
    '_open', req)
  File "/usr/lib/python3.5/urllib/request.py", line 444, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.5/urllib/request.py", line 1297, in https_open
    context=self._context, check_hostname=self._check_hostname)
  File "/usr/lib/python3.5/urllib/request.py", line 1256, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [Errno 110] Connection timed out>
"No such interface 'org.freedesktop.DBus.Properties' on object at path /"
QNetworkDiskCache: couldn't remove the cache file  "/home/shevek/.cache/scudcloud/data8/6/16ba9wo6.d"
QNetworkDiskCache: couldn't remove the cache file  "/home/shevek/.cache/scudcloud/data8/1/yqbl1qjq.d"
"No such interface 'org.freedesktop.DBus.Properties' on object at path /"
"No such interface 'org.freedesktop.DBus.Properties' on object at path /"

```

## Steps to reproduce
 1. Nothing special.
 2. 
 3.

**Comment by raelgc** [comment]
Thanks for reporting this issue, @shevek!

Is this happening everytime at startup? Or in some conditions, like when viewing a snippet?

**Comment by shevek** [comment]
It's not at startup. It's just something that happens "sometimes", perhaps related to network instability. One network I was using had 20% packet loss until today, so I got a lot of issues. It can also go wrong when resuming from sleep, depending on when the sleep interrupt happened.

**Comment by shevek** [comment]
After latest restart, console just contains:
```
SpellCheck: Language = "en_US"
"No such interface 'org.freedesktop.DBus.Properties' on object at path /"
"No such interface 'org.freedesktop.DBus.Properties' on object at path /org/freedesktop/NetworkManager/ActiveConnection/22"
QNetworkReplyImplPrivate::error: Internal problem, this method must only be called once.
"No such interface 'org.freedesktop.DBus.Properties' on object at path /"
"No such interface 'org.freedesktop.DBus.Properties' on object at path /"
"No such interface 'org.freedesktop.DBus.Properties' on object at path /"
```

I don't understand dbus - could this be asynchronicity in suspend/resume?

**Comment by shevek** [comment]
QDBusObjectPath: invalid path ""
QDBusConnection: error: could not send message to service "org.freedesktop.NetworkManager" path "/org/freedesktop/NetworkManager" interface "org.freedesktop.NetworkManager" member "ActivateConnection": Marshalling failed: Invalid object path passed in arguments
QDBusObjectPath: invalid path ""
QDBusConnection: error: could not send message to service "org.freedesktop.NetworkManager" path "/org/freedesktop/NetworkManager" interface "org.freedesktop.NetworkManager" member "ActivateConnection": Marshalling failed: Invalid object path passed in arguments
QNetworkReplyImplPrivate::error: Internal problem, this method must only be called once.
QNetworkReplyImplPrivate::error: Internal problem, this method must only be called once.
QNetworkReplyImplPrivate::error: Internal problem, this method must only be called once.
QNetworkReplyImplPrivate::error: Internal problem, this method must only be called once.
QNetworkReplyImplPrivate::error: Internal problem, this method must only be called once.
QNetworkReplyImplPrivate::error: Internal problem, this method must only be called once.
QNetworkReplyImplPrivate::error: Internal problem, this method must only be called once.
QNetworkReplyImplPrivate::error: Internal problem, this method must only be called once.
QNetworkReplyImplPrivate::error: Internal problem, this method must only be called once.
QNetworkReplyImplPrivate::error: Internal problem, this method must only be called once.
QNetworkReplyImplPrivate::error: Internal problem, this method must only be called once.
"No such interface 'org.freedesktop.DBus.Properties' on object at path /"
"No such interface 'org.freedesktop.DBus.Properties' on object at path /"

**Comment by shevek** [comment]
and it's not working. GUI's responding, but it isn't _working_

**Comment by raelgc** [comment]
Unfortunatelly, all above are just simple console output, and not crash exceptions. These days, as I'm moving to a new home (so I'm under a slow 3G connection), I'm facing these outputs too.

For some reason, Qt5 appears to not handle so well the slow connections as Qt4.

Anyway, I'll investigate all issues.

**Comment by shevek** [comment]
Perversely, I'm glad about the 3G, as it's much easier to debug when you can reproduce! But yes, I'm on a network with 20% packet loss, and scudcloud is one of the major losers. Firefox often manages to hold its own. mpv sometimes works. Skype works OK, except when it doesn't.

**Comment by raelgc** [comment]
Fixed in commit 68d7156
